improver/utilities/interpolation.py

```diff
@@ -84,11 +91,126 @@ def interpolate_missing_data(
 
     if limit is not None:
         index = ~np.isfinite(data) & np.isfinite(data_filled)
-        data_filled_above_limit = (data_filled[index] > limit[index])
-        index[index] = data_filled_above_limit
-        data_filled[index] = limit[index]
+        limit_data_values(data_filled, index, limit, limit_as_maximum)
 
     index = ~np.isfinite(data)
     data[index] = data_filled[index]
 
     return data
+
+
+def limit_data_values(data, index, limit, limit_as_maximum):
+    """
+    Impose a limit on data.
+    """
+    index = index.copy()
+    if limit_as_maximum:
+        data_violating_limit = (data[index] > limit[index])
+    else:
+        data_violating_limit = (data[index] < limit[index])
+    index[index] = data_violating_limit
+    data[index] = limit[index]
+
+
+class InterpolateUsingDifference:
+    """
+    Uses interpolation to fill holes in the data contained within the input
+    cube. This is achieved by calculating the difference between the input cube
+    and a complete (filling the whole domain) reference cube. The difference
```

Comment:
I've changed it to something maybe clearer. Again, I am trying to be a bit general, so I don't want to say 'filled in' as it could be used with a field that is simply always continuous.